server/protocols/webadmin/webadmin-cassandra/src/test/java/org/apache/james/webadmin/routes/CassandraMigrationRoutesTest.java

```diff
@@ -0,0 +1,202 @@
+            .statusCode(200)
+            .body(is("{\"version\":" + LATEST_VERSION + "}"));
+    }
+
+    @Ignore
+    @Test
+    public void postShouldReturnConflictWhenMigrationOnRunning() throws Exception {
+        when()
+            .post("/upgrade")
+        .then()
+            .statusCode(409);
+    }
+
+    @Test
+    public void postShouldReturnErrorCodeWhenInvalidVersion() throws Exception {
+        when(schemaVersionDAO.getCurrentSchemaVersion()).thenReturn(CompletableFuture.completedFuture(Optional.of(OLDER_VERSION)));
+
+        given()
+            .body(String.valueOf("NonInt"))
+        .with()
+            .post("/upgrade")
+        .then()
+            .statusCode(400);
+
+        verifyNoMoreInteractions(schemaVersionDAO);
+    }
+
+    @Test
+    public void postShouldDoMigrationToNewVersion() throws Exception {
+        when(schemaVersionDAO.getCurrentSchemaVersion()).thenReturn(CompletableFuture.completedFuture(Optional.of(OLDER_VERSION)));
+
+        given()
+            .body(String.valueOf(CURRENT_VERSION))
+        .with()
+            .post("/upgrade")
+        .then()
+            .statusCode(204);
+
+        verify(schemaVersionDAO, times(1)).getCurrentSchemaVersion();
+        verify(schemaVersionDAO, times(1)).updateVersion(eq(CURRENT_VERSION));
+        verifyNoMoreInteractions(schemaVersionDAO);
+    }
+
+    @Test
+    public void postShouldNotDoMigrationWhenCurrentVersionIsNewerThan() throws Exception {
+        when(schemaVersionDAO.getCurrentSchemaVersion()).thenReturn(CompletableFuture.completedFuture(Optional.of(CURRENT_VERSION)));
+
+        given()
+            .body(String.valueOf(OLDER_VERSION))
+        .with()
+            .post("/upgrade")
+        .then()
+            .statusCode(410);
+
+        verify(schemaVersionDAO, times(1)).getCurrentSchemaVersion();
+        verifyNoMoreInteractions(schemaVersionDAO);
+    }
+
+    @Test
+    public void postShouldDoMigrationToLatestVersion() throws Exception {
+        when(schemaVersionDAO.getCurrentSchemaVersion()).thenReturn(CompletableFuture.completedFuture(Optional.of(OLDER_VERSION)));
+
+        when()
+            .post("/upgrade/latest")
+        .then()
+            .statusCode(200);
+
+        verify(schemaVersionDAO, times(1)).getCurrentSchemaVersion();
+        verify(schemaVersionDAO, times(1)).updateVersion(eq(CURRENT_VERSION));
+        verify(schemaVersionDAO, times(1)).updateVersion(eq(LATEST_VERSION));
+        verifyNoMoreInteractions(schemaVersionDAO);
+    }
+
+    @Test
+    public void postShouldNotDoMigrationToLatestVersionWhenItIsUpToDate() throws Exception {
+        when(schemaVersionDAO.getCurrentSchemaVersion()).thenReturn(CompletableFuture.completedFuture(Optional.of(LATEST_VERSION)));
+
+        when()
+            .post("/upgrade/latest")
+        .then()
+            .statusCode(410);
+
+        verify(schemaVersionDAO, times(1)).getCurrentSchemaVersion();
+        verifyNoMoreInteractions(schemaVersionDAO);
+    }
```

Comment:
A current version is always returned, and is defined in CassandraSchemaVersionManager